utils/src/metrics.rs

```diff
@@ -0,0 +1,98 @@
+use prometheus_client::encoding::EncodeLabelSet;
+use prometheus_client::metrics::counter::Counter;
+use prometheus_client::metrics::family::Family;
+
+/// Metric label that indicates success or failure.
+#[derive(Clone, Debug, Hash, PartialEq, Eq, EncodeLabelSet)]
+pub struct Status {
+    pub ok: &'static str,
```

Comment:
Usually, the way I've seen this done is instead tracking "success" and "total" count (computing `failed` as the delta).

This is often because you typically want "total" anyways and I believe 2 counters are more efficient than this. But I don't have any proof that is true (just what is more canonical).

I think you could instead have `Status` reference two independent counters of the same prefix.

Reply:
After some reflection, I think the `Family<>` will make this easier to work with/track than my suggestion.